openwisp_monitoring/check/tests/test_ping.py

```diff
@@ -110,8 +110,8 @@ def test_device_deleted(self):
         check.save()
         # dev = d
         # import ipdb; ipdb.set_trace()
-        d.delete()
         check = Check.objects.get(pk=check.pk)
+        d.delete()
```

Comment:
I now understood why you had to do this.
Now the related check will be deleted and hence the get query will fail.

The test written here was before passing but it was wrong.

Let's improve this part:

- this test doesn't have reason to be anymore, especially here in the `test_ping` file, 
- let's create a new test in `test_models` with the following code, which fails correctly before applying your patch:

```python
def test_device_deleted(self):
        self.assertEqual(Check.objects.count(), 0)
        d = self._create_device(organization=self._create_org())
        self.assertEqual(Check.objects.count(), 1)
        d.delete()
        self.assertEqual(Check.objects.count(), 0)
```

Reply:
I now understand why the `Check` was earlier getting deleted and what this test was checking out. There's a check in `_get_ip()` which was deleting the Check if the device had been deleted :blush:. I think that check can be safely removed as the function won't get triggered now due to Check being deleted.